backend/tests/data_generator.py

```diff
@@ -47,6 +48,7 @@ def gen_user(
     def gen_artefact(
         self,
         stage_name: str,
+        family_name: str = "snap",
```

Comment:
This change resulted in a bunch of propagation. Necessary because the testing utilities derived family from the stage, but Charms share stage names with Snaps so that is no longer possible.